Source/Scene/Model.js

```diff
@@ -2893,11 +2900,46 @@ function createTexture(gltfTexture, model, context) {
     wrapS === TextureWrap.MIRRORED_REPEAT ||
     wrapT === TextureWrap.REPEAT ||
     wrapT === TextureWrap.MIRRORED_REPEAT;
-
+  var npot;
   var tx;
   var source = gltfTexture.image;
 
   if (defined(internalFormat)) {
+    npot =
+      !CesiumMath.isPowerOfTwo(gltfTexture.width) ||
+      !CesiumMath.isPowerOfTwo(gltfTexture.height);
+
+    // Warning to encourage power of 2 texture dimensions with KHR_texture_basisu
+    if (npot && PixelFormat.isCompressedFormat(internalFormat)) {
+      console.warn(
+        "For maximim compatibility, we encourage power of 2 dimensions when using compressed textures in glTF. See the Model.js constructor documentation for more information."
+      );
+    }
+
+    if (
+      !defined(gltfTexture.mipLevels) &&
+      (minFilter === TextureMinificationFilter.NEAREST_MIPMAP_NEAREST ||
+        minFilter === TextureMinificationFilter.NEAREST_MIPMAP_LINEAR)
+    ) {
+      sampler = new Sampler({
+        wrapS: sampler.wrapS,
+        wrapT: sampler.wrapT,
+        minificationFilter: TextureMinificationFilter.NEAREST,
+        magnificationFilter: sampler.magnificationFilter,
+      });
+    } else if (
+      !defined(gltfTexture.mipLevels) &&
+      (minFilter === TextureMinificationFilter.LINEAR_MIPMAP_NEAREST ||
+        minFilter === TextureMinificationFilter.LINEAR_MIPMAP_LINEAR)
+    ) {
+      sampler = new Sampler({
+        wrapS: sampler.wrapS,
+        wrapT: sampler.wrapT,
+        minificationFilter: TextureMinificationFilter.LINEAR,
+        magnificationFilter: sampler.magnificationFilter,
+      });
+    }
```

Comment:
I feel like this code could be more concise. Maybe `minificationFilter` can get set first, and then the sampler can get created.